taxcalc/tests/test_functions.py

```diff
@@ -0,0 +1,40 @@
+"""
+Tests for Tax-Calculator functions.py file.
+"""
+# CODING-STYLE CHECKS:
+# pep8 --ignore=E402 test_functions.py
+# pylint --disable=locally-disabled test_functions.py
+# (when importing numpy, add "--extension-pkg-whitelist=numpy" pylint option)
+
+import os
+import sys
+import re
+CUR_PATH = os.path.abspath(os.path.dirname(__file__))
+
+
+def test_function_args_usage():
+    """
+    Checks each function argument in functions.py for use in its function body.
+    """
+    funcfilename = '{}/../functions.py'.format(CUR_PATH)
+    with open(funcfilename, 'r') as funcfile:
```

Comment:
To go down this path, I would suggest making use of the `inspect` module, documented [here](https://docs.python.org/2/library/inspect.html) - in particular, `inspect.getsource`. I do have a concern about this being a somewhat fragile test, not because I have a concrete counter example, but rather just that having a test pass based on matching a regular expression to the source code of a function makes me somewhat nervous. I suspect someone could do something very innocuous, end up with a failed test, and have a very difficult to understand error message.